app/src/test/java/nerd/tuxmobil/fahrplan/congress/schedule/LectureViewDrawerTest.kt

```diff
@@ -0,0 +1,227 @@
+
+        assertThat(layoutParams.size).isEqualTo(1)
+        assertThat(layoutParams[lectures.first()]).isNotNull()
+    }
+
+    @Test
+    fun `calculateLayoutParams for single lecture returns margins 0`() {
+        val lectures = listOf(createLecture(_roomIndex = 0))
+        conference.calculateTimeFrame(lectures) { Moment(it).minuteOfDay }
+
+        val layoutParams = LectureViewDrawer.calculateLayoutParams(0, lectures, standardHeight, conference)
+        val lectureParams = layoutParams[lectures.first()]
+
+        assertMargins(lectureParams, 0, 0)
+    }
+
+    @Test
+    fun `calculateLayoutParams for single UTC lecture sets top margin 0 (its the first lecture in all rooms, so on the top)`() {
+        val startTime = 10 * 60 // 10:00am
+        val lectures = listOf(createLecture(_roomIndex = 0, date = conferenceDate, time = startTime))
+        conference.calculateTimeFrame(lectures) { dateUTC: Long -> Moment(dateUTC).minuteOfDay }
+
+        val layoutParams = LectureViewDrawer.calculateLayoutParams(0, lectures, standardHeight, conference)
+        val lectureParams = layoutParams[lectures.first()]
+
+        assertMargins(lectureParams, 0, 0)
+    }
+
+    @Test
+    fun `calculateLayoutParams for single *none* UTC lecture sets top margin 0 (its the first lecture in all rooms, so on the top)`() {
+        val startTime = 10 * 60 // 10:00am
+        val lectures = listOf(createLecture(_roomIndex = 0, time = startTime))
+        conference.calculateTimeFrame(lectures) { dateUTC: Long -> Moment(dateUTC).minuteOfDay }
+
+        val layoutParams = LectureViewDrawer.calculateLayoutParams(0, lectures, standardHeight, conference)
+        val lectureParams = layoutParams[lectures.first()]
+
+        assertMargins(lectureParams, 0, 0)
+    }
+
+    @Test
+    fun `calculateLayoutParams for consecutive lecture sets margins based on gap duration`() {
+        val startTime1 = 10 * 60 // 10:00am
+        val duration1 = 45
+        val gapMinutes = 15
+        val startTime2 = startTime1 + duration1 + gapMinutes // 11:00am
+
+        val lecture1 = createLecture(_roomIndex = 0, date = conferenceDate, time = startTime1, _duration = duration1)
+        val lecture2 = createLecture(_roomIndex = 0, date = conferenceDate, time = startTime2)
+        val lectures = listOf(lecture1, lecture2)
+        conference.calculateTimeFrame(lectures) { dateUTC: Long -> Moment(dateUTC).minuteOfDay }
+
+        val layoutParams = LectureViewDrawer.calculateLayoutParams(0, lectures, standardHeight, conference)
+        val lecture1Params = layoutParams[lecture1]
+        val secondLectureParams = layoutParams[lecture2]
+
+        assertMargins(lecture1Params, 0, gapMinutes)
+        assertMargins(secondLectureParams, 0, 0)
+    }
+
+    @Test
+    fun `calculateLayoutParams for consecutive lecture in another room sets top margin based on gap duration`() {
```

Comment:
:question: Why does a `lecture0` in `room0` influence `lecture1` in `room1`?

Reply:
the wording is misleading. I changed it to `.... sets top margin based on conference day start`

Reply:
Finally, I understand the test. It checks that lecture2 is pushed down with a 1h marginTop because lecture1 defines the conference start at 10:00. This took me a while - actually until I was about to ask you once again. The rubberduck effect. Thanks for your patience.